src/main/java/duke/model/events/DoWithin.java

```diff
@@ -0,0 +1,32 @@
+package duke.model.events;
+
+import java.time.LocalDateTime;
+
+public class DoWithin extends TaskWithDates {
+    private LocalDateTime endDate;
+
+    /**
+     * Initializes a task to be done within two date the given description.
+     *
+     * @param description A description of this task.
+     * @param startDate Starting date of task
+     * @param endDate Ending date of task
+     */
+    public DoWithin(String description, LocalDateTime startDate, LocalDateTime endDate) {
+        super(description, startDate);
+        this.endDate = endDate;
+    }
+
+    @Override
+    public String toString() {
+        return ("[W]" + super.toString() + " within " + super.getStartDate() + " to " + this.endDate);
```

Comment:
why not use this in the storage as well?

Reply:
Code deleted.